tests/common/get_group_range.h

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2017-2022 Codeplay Software LTD. All Rights Reserved.
+//  Copyright (c) 2020-2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_TESTS_COMMON_GET_GROUP_RANGE_H
+#define __SYCLCTS_TESTS_COMMON_GET_GROUP_RANGE_H
+
+namespace sycl_cts {
+namespace util {
+/**
+ * Since \p sycl::range by standard provides no default constructor,
+ * this function returns a \p sycl::range with 1 for each dimension. */
+template <int Dimensions>
+sycl::range<Dimensions> get_default_range();
+
+template <>
+inline sycl::range<1> get_default_range() {
+  return sycl::range<1>{1};
+}
+
+template <>
+inline sycl::range<2> get_default_range() {
+  return sycl::range<2>{1, 1};
+}
+
+template <>
+inline sycl::range<3> get_default_range() {
+  return sycl::range<3>{1, 1, 1};
+}
+
+/**
+ * @brief Provides range for maximal size work-group
+ *        supported by device of a given queue.
+ *        Multidimentional work-group range is made
+ *        as hyper-cybic as possible
+ * @tparam Dimensions Dimension to use for group instance
+ */
+template <int Dimensions>
+sycl::range<Dimensions> work_group_range(
+    sycl::queue queue,
+    uint64_t work_items_limit = std::numeric_limits<uint64_t>::max()) {
```

Comment:
Shouldn't we use `size_t` here? `max_work_group_size` is `size_t`.

Reply:
You're right. Fixed it.